components/admin/home/update-event-popup.vue

```diff
@@ -0,0 +1,115 @@
+<script setup lang="ts">
+import dayjs from 'dayjs'
+import type { EventWithAttendees, Event } from '~/shared/types'
+import { useMutation, useQueryClient } from '@tanstack/vue-query'
+import { ref } from 'vue'
+
+const props = defineProps<{
+  event: EventWithAttendees
+  showPopup: boolean
+}>()
+const queryClient = useQueryClient()
+
+const mutation = useMutation({
+    mutationFn: (id: string) => $api(`/api/event/${id}`, {
+        method: 'PUT',
+        body: changedEvent.value,
+    }),
+})
+const msg = ref('')
+
+const closePopup = defineEmits(['closePopup'])
+
+const state = reactive({
+  ...props.event,
+  startDateTime: dayjs(parseInt(props.event.startDateTime) * 1000).format("YYYY-MM-DDTHH:mm"),
+  endDateTime: dayjs(parseInt(props.event.endDateTime) * 1000).format("YYYY-MM-DDTHH:mm"),
+})
+const changedEvent = ref<Partial<Event>>({})
+
+const handleSubmit = () => {
+  // Check differences between original and edited
+  for (const key in props.event) {
+    if (key === "startDateTime" || key === "endDateTime") {
+      if (parseInt(props.event[key as keyof Event]) !== dayjs(state[key as keyof Event]).unix()) {
+        changedEvent.value[key as keyof Event] = state[key as keyof Event]
+      }
+    } else {
+      if (props.event[key as keyof Event] !== state[key as keyof Event]) {
+        changedEvent.value[key as keyof Event] = state[key as keyof Event]
+      }
+    }
+  }
```

Comment:
Why do we need to check diff? Seems like a lot of complexity if we can just send and replace data on server